### PR TITLE
Removing SElinux policy as BuildRequire

### LINKED ISSUE #631: [BUG] Conflicting se-linux policy version error when installing microshift
#### What happened: 

I encountered a "Problem: conflicting requests" error when installing microshift (rpm-ostree install microshift)

"cannot install both selinux-policy-35.15-1.fc35.noarch and selinux-policy-35.7-1.fc35.noarch"

#### What you expected to happen: 

I expected the installation to succeed

#### How to reproduce it (as minimally and precisely as possible):

1.  I followed the procedure documented here: https://www.opensourcerers.org/2022/01/17/openshift-on-raspberry-pi-4/
2. Issue was encountered when running rpm-ostree install microshift

#### Environment:
- Microshift version (use `microshift version`): 4.8.0
- Hardware configuration:  Raspberry Pi 3 
- OS (e.g: `cat /etc/os-release`): Fedora Linux 35.20220101.0 (IoT Edition)
- Kernel (e.g. `uname -a`): 5.15.12-200.fc35.aarch64
- Others:

#### Relevant Logs 

[root@pi1 ~]# rpm-ostree install microshift
Checking out tree 0cbf945... done
Enabled rpm-md repositories: updates-modular copr:copr.fedorainfracloud.org:group_redhat-et:microshift fedora-modular updates fedora fedora-cisco-openh264
Updating metadata for 'updates-modular'... done
Updating metadata for 'updates'... done
Importing rpm-md... done
rpm-md repo 'updates-modular'; generated: 2022-03-11T14:51:11Z solvables: 1365
rpm-md repo 'copr:copr.fedorainfracloud.org:group_redhat-et:microshift' (cached); generated: 2022-03-11T14:28:08Z solvables: 8
rpm-md repo 'fedora-modular' (cached); generated: 2021-10-26T05:06:57Z solvables: 1263
rpm-md repo 'updates'; generated: 2022-03-11T14:01:58Z solvables: 25395
rpm-md repo 'fedora' (cached); generated: 2021-10-26T05:31:21Z solvables: 56722
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2021-09-21T18:07:30Z solvables: 4
Resolving dependencies... done
error: Could not depsolve transaction; 1 problem detected:
 Problem: conflicting requests
  - package microshift-4.8.0-2022_03_08_195111.fc35.aarch64 requires microshift-selinux, but none of the providers can be installed
  - package microshift-4.8.0-2022_03_11_124751.fc35.aarch64 requires microshift-selinux, but none of the providers can be installed
  - package microshift-selinux-4.8.0-2022_03_08_195111.fc35.noarch requires selinux-policy >= 35.15-1.fc35, but none of the providers can be installed
  - package microshift-selinux-4.8.0-2022_03_08_195111.fc35.noarch requires selinux-policy-base >= 35.15-1.fc35, but none of the providers can be installed
  - package microshift-selinux-4.8.0-2022_03_11_124751.fc35.noarch requires selinux-policy >= 35.15-1.fc35, but none of the providers can be installed
  - package microshift-selinux-4.8.0-2022_03_11_124751.fc35.noarch requires selinux-policy-base >= 35.15-1.fc35, but none of the providers can be installed
  - cannot install both selinux-policy-35.15-1.fc35.noarch and selinux-policy-35.7-1.fc35.noarch

### PR DESCRIPTION
**Pull request by oglok**
Signed-off-by: Ricardo Noriega <rnoriega@redhat.com>

**Which issue(s) this PR addresses**:

Closes #631

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
Removing the `selinux-policy-devel` BR means the SELinux macros go away. You definitely don't want that.

File: packaging/rpm/microshift.spec [hunk 1]
```diff
@@ -95,11 +94,8 @@ Things break. But you can still help shape it, too.
 
 %package selinux
 Summary: SELinux policies for MicroShift
-BuildRequires: selinux-policy >= %{selinux_policyver}
-BuildRequires: selinux-policy-devel >= %{selinux_policyver}
 Requires: container-selinux >= %{container_policy_epoch}:%{container_policyver}
 BuildArch: noarch
-%{?selinux_requires}
```
Comment (mangelajo) [inline packaging/rpm/microshift.spec:102]: ^ but if we remove this, the selinux dependencies aren't added anymore, right?

File: packaging/rpm/microshift.spec [hunk 2]
```diff
@@ -95,11 +94,8 @@ Things break. But you can still help shape it, too.
 
 %package selinux
 Summary: SELinux policies for MicroShift
-BuildRequires: selinux-policy >= %{selinux_policyver}
-BuildRequires: selinux-policy-devel >= %{selinux_policyver}
```
Comment (mangelajo) [inline packaging/rpm/microshift.spec:99]: Are we sure this works now on RHEL8? it was added because of that platform. We can try...

**Comment by mangelajo** [comment]
> Your actual issue is that FCOS and FIoT don't keep up on SELinux policy stuff, so you probably want to drop `%{?selinux_requires}`.

 I'm not sure that will work.

If we compile newer policies, and then install on an outdated selinux system, those policies won't load.

I believe that probably forcing users to update first the selinux packages, and then, installing microshift, is the right thing to do, and not a workaround.

**Comment by Conan-Kudo** [comment]
Yes, you're right it is, but removing the devel package will break compiling SELinux policy modules and remove the macros.

File: packaging/rpm/microshift.spec (hunk 2 above)
Comment (oglok) [reply packaging/rpm/microshift.spec:99]:
```
[vagrant@rhel8 microshift]$ cat packaging/rpm/_rpmbuild/SPECS/microshift.spec | grep -A5 SELinux 
# SELinux specifics
%global selinuxtype targeted
%define container_policyver 2.167.0-1
%define container_policy_epoch 2
%define microshift_relabel_files() \
   mkdir -p /var/hpvolumes; \
--
Summary: SELinux policies for MicroShift
Requires: container-selinux >= %{container_policy_epoch}:%{container_policyver}
BuildArch: noarch

%description selinux
SElinux policy modules for MicroShift.
--
# SELinux modules build

cd packaging/selinux
make

%install
```

After `make rpm`

```
[vagrant@rhel8 microshift]$ sudo dnf install packaging/rpm/_rpmbuild/RPMS/noarch/microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch.rpm packaging/rpm/_rpmbuild/RPMS/x86_64/microshift-4.8.0-202
2_03_11_124751_6_gcf34c42a.el8.x86_64.rpm                                                                                                                                                                                                               
Updating Subscription Management repositories.
Last metadata expiration check: 1:54:21 ago on Tue 29 Mar 2022 10:17:08 AM UTC.                                                                                                                                
Dependencies resolved.
===============================================================================================================================================================================================================                                         
 Package                                          Architecture                         Version                                                                Repository                                  Size                                          
===============================================================================================================================================================================================================                                         
Installing:        
 microshift                                       x86_64                               4.8.0-2022_03_11_124751_6_gcf34c42a.el8                                @commandline                                36 M                                          
 microshift-selinux                               noarch                               4.8.0-2022_03_11_124751_6_gcf34c42a.el8                                @commandline                                19 k                                          
                                                   
Transaction Summary     
===============================================================================================================================================================================================================                                         
Install  2 Packages
                                                                                                                                                                                                               
Total size: 36 M                                                                                                                                                                                               
Installed size: 154 M                                                                                                                                                                                          
Is this ok [y/N]: y                                                                                                                                                                                            
Downloading Packages:                                                                                                                                                                                          
Running transaction check                                                                                                                                                                                      
Transaction check succeeded.                                                                                                                                                                                   
Running transaction test                                                                                                                                                                                       
Transaction test succeeded.                                                                                                                                                                                    
Running transaction        
  Preparing        :                                                                                                                                                                                       1/1                                          
  Installing       : microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch                                                                                                                     1/2                                          
  Running scriptlet: microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch                                                                                                                     1/2 
  Installing       : microshift-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.x86_64                                                                                                                             2/2                                          
  Running scriptlet: microshift-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.x86_64                                                                                                                             2/2                                          
  Running scriptlet: microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch                                                                                                                     2/2                                          
  Running scriptlet: microshift-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.x86_64                                                                                                                             2/2                                          
  Verifying        : microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch                                                                                                                     1/2                                          
  Verifying        : microshift-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.x86_64                                                                                                                             2/2                                          
Installed products updated.                                                                            
                                                   
Installed:
  microshift-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.x86_64                                          microshift-selinux-4.8.0-2022_03_11_124751_6_gcf34c42a.el8.noarch                                                                                  
                                                   
Complete!         
```

Starting the systemd service, and after a while:

```
[vagrant@rhel8 microshift]$ oc get pods -A
NAMESPACE                       NAME                                  READY   STATUS    RESTARTS   AGE
kube-system                     kube-flannel-ds-68pwq                 1/1     Running   0          2m30s
kubevirt-hostpath-provisioner   kubevirt-hostpath-provisioner-7sxkt   1/1     Running   0          2m10s
openshift-dns                   dns-default-gfz2t                     2/2     Running   0          2m30s
openshift-dns                   node-resolver-vs55f                   1/1     Running   0          2m30s
openshift-ingress               router-default-6c96f6bc66-6t47v       1/1     Running   0          2m34s
openshift-service-ca            service-ca-7bffb6f6bf-vlrks           1/1     Running   0          2m35s
```

```
 cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.4 (Ootpa)
```

**Pull request closed without merging**